eth/tracers/js/goja.go

```diff
@@ -258,7 +254,7 @@ func (t *jsTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, scope
 	if !t.traceStep {
 		return
 	}
-	if t.err != nil {
+	if t.err != nil || err != nil {
```

Comment:
We should let the JS code decide what to do in case of error IMO.
```suggestion
	if t.err != nil {
```